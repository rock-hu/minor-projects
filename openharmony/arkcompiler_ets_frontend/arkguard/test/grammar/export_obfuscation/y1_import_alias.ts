/*
 * Copyright (c) 2024 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import assert from 'assert';

export * as CDE1 from './y1_export_alias';
export { default } from './y1_export_alias';
import * as CDE2 from './y1_export_alias';

let ins = new CDE2.ClassAlis;
assert(ins.classProp1 === 2, 'success');
type typeAlis = CDE2.InterOriginal;
assert(CDE2.fooOriginal() === '11', 'success');

