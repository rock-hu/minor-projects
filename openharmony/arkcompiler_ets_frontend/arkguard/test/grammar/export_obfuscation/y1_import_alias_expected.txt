/*
 * Copyright (c) 2024 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import assert from 'assert';
export * as t5 from './y1_export_alias';
export { default } from './y1_export_alias';
import * as q5 from './y1_export_alias';
let r5 = new q5.o5;
assert(r5.p5 === 2, 'success');
type s5 = q5.m5;
assert(q5.k5() === '11', 'success');